flink-autoscaler-standalone/src/main/java/org/apache/flink/autoscaler/standalone/StandaloneAutoscalerExecutor.java

```diff
@@ -75,29 +98,38 @@ public void start() {
     @Override
     public void close() {
         scheduledExecutorService.shutdownNow();
+        scalingThreadPool.shutdownNow();
     }
 
     @VisibleForTesting
     protected void scaling() {
         LOG.info("Standalone autoscaler starts scaling.");
         try {
             var jobList = jobListFetcher.fetch();
+            Collection<Future<?>> futures = new LinkedList<>();
             for (var jobContext : jobList) {
-                try {
-                    autoScaler.scale(jobContext);
-                } catch (Throwable e) {
-                    LOG.error("Error while scaling job", e);
-                    eventHandler.handleEvent(
-                            jobContext,
-                            AutoScalerEventHandler.Type.Warning,
-                            AUTOSCALER_ERROR,
-                            e.getMessage(),
-                            null,
-                            null);
-                }
+                futures.add(scalingThreadPool.submit(() -> scalingSingleJob(jobContext)));
+            }
+            for (Future<?> future : futures) {
+                future.get();
             }
         } catch (Throwable e) {
             LOG.error("Error while fetch job list.", e);
         }
```

Comment:
```suggestion
        } catch (Throwable e) {
            LOG.error("Error while executing autoscaling.", e);
        }
```

Note that none of the individual scheduled futures have error handling here. I guess that is ok because the contract is that the autoscaler handles all exceptions, so this is just a safe-guard to return a better error message.

Reply:
From current code, `scalingThreadPool` execute the `scalingSingleJob` logic, and `scalingSingleJob` catches all exceptions. So I think all exceptions of control loop thread are from `fetch job list`.

WDYT?

Reply:
If you are confident about this, how about moving the try/catch to be around the `.fetch()` method?

Reply:
Updated.